### PR TITLE
Remove case sensitivity from filtering in tables, fix date sorting in tables

### PR DESCRIPTION
**Pull request by William-Unosquare**
- Fixes errors when sorting dates in tables.
- Fixes searching for some things in tables being case sensitive.

### CODE REVIEW

File: src/components/DataTable/Cells/contracts.js [hunk 1]
```diff
@@ -19,18 +25,26 @@ const endDate = {
     }
     return endDate.format('Do MMMM YYYY');
   },
+  sortMethod: (a, b) => (moment(a).isBefore(moment(b)) ? 1 : -1),
```
Comment (d1wilko) [inline src/components/DataTable/Cells/contracts.js]: could this function be assigned to a variable - duplication on line 10 and line 28

File: src/components/DataTable/Cells/contracts.js [hunk 2]
```diff
@@ -7,6 +7,12 @@ const startDate = {
     const startDate = new moment(contract.startDate, 'YYYY-MM-DD');
     return startDate.format('Do MMMM YYYY');
   },
+  sortMethod: (a, b) => (moment(a).isBefore(moment(b)) ? 1 : -1),
+  filterMethod: ({ value }, { startDate }) =>
```
Comment (d1wilko) [inline src/components/DataTable/Cells/contracts.js]: probably should have more self explanatory arg names than a and b?

File: src/components/DataTable/Cells/holidays.js [hunk 3]
```diff
@@ -34,7 +35,7 @@ const startDate = {
   accessor: holiday => holiday.eventDates[0].startDate,
 
   Cell: cell => moment(cell.row.startDate).format('Do MMM YYYY'),
-  sortMethod: (a, b) => (a.isBefore(b) ? 1 : -1),
+  sortMethod: (a, b) => (moment(a).isBefore(moment(b)) ? 1 : -1),
```
Comment (d1wilko) [inline src/components/DataTable/Cells/holidays.js]: this smells like a helper or something

File: src/components/DataTable/Cells/holidays.js [hunk 4]
```diff
@@ -48,7 +49,7 @@ const endDate = {
   accessor: holiday =>
     holiday.eventDates[holiday.eventDates.length - 1].endDate,
   Cell: cell => moment(cell.row.endDate).format('Do MMM YYYY'),
-  sortMethod: (a, b) => (a.isBefore(b) ? 1 : -1),
+  sortMethod: (a, b) => (moment(a).isBefore(moment(b)) ? 1 : -1),
```
Comment (d1wilko) [inline src/components/DataTable/Cells/holidays.js]: i see duplication again

**Review by d1wilko — CHANGES_REQUESTED** [review]
alot of code duplication going on even with the filter methods

File: src/components/DataTable/Cells/clients.js [hunk 5]
```diff
@@ -2,6 +2,7 @@ const clientName = {
   id: 'clientName',
   Header: 'Client',
   accessor: client => client.clientName,
+  filterMethod: ({ value }, { clientName }) => clientName.toLowerCase().includes(value.toLowerCase()),
```
Comment (BrownMark) [inline src/components/DataTable/Cells/clients.js]: Seeing a lot of repetition here, again some technical debt to make note of just.

**Review by BrownMark — APPROVED** [review]
Approved

**Comment by William-Unosquare** [comment]
Removed the duplication by making sort and filter utility functions

**Review by d1wilko — APPROVED** [review]
nice - cheers for that :)

**Pull request merged**